Weather App Completion

ershadul1 opened the pull request:
### What does this PR do?

- [X] Builds a weather app using open weather API.
- [X] Uses webpack to bundle Javascript modules. 
- [X] Builds a responsive webpage with industry-standard design.
- [X] Changes background according to the weather condition.
- [X] Follows coding best practices.

ershadul1 committed:
Add starter files

ershadul1 committed:
Add .vscode to gitignore

ershadul1 committed:
Delete settings.json

ershadul1 committed:
Add form for user input

ershadul1 committed:
Add event listener for search btn

ershadul1 committed:
Add radio buttons to switch temp scale

ershadul1 committed:
Add removeChildren method in helper

ershadul1 committed:
Add more containers in index.js

ershadul1 committed:
Add getWeather async function

ershadul1 committed:
Add populate weather data method

ershadul1 committed:
Add method for converting date

ershadul1 committed:
Add invalid input to warn user about invalid input

ershadul1 committed:
Add get background  func to get an image

ershadul1 committed:
Add change-background method

ershadul1 committed:
Update styling of the project

ershadul1 committed:
Build using webpack

ershadul1 committed:
Merge branch 'weather-app' of github.com:ershadul1/weather-now into weather-app

ershadul1 committed:
Fix open weather url

ershadul1 committed:
Center weather data container

ershadul1 committed:
Fix typo

ershadul1 committed:
Update README.md

ershadul1 committed:
Update live demo link

Forison commented on package.json:
```diff
@@ -0,0 +1,44 @@
+  "description": "A simple weather app that fetches weather data using open weather API, according to user input, and displays it to the user. Built with javascript.",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
```
- let us add the dev to our script
```
"scripts": {
    "dev": "webpack --mode development",
    "start": "webpack-dev-server --mode development --open" // optional
  },
```

Forison commented on src/modules/get-background.js:
```diff
@@ -0,0 +1,19 @@
+    const response = await fetch(`https://pixabay.com/api/?key=${apiKey}&q=${weatherType}+sky&image_type=photo&pretty=true&editors_choice=true`);
+    const background = await response.json();
+    if (background.cod === '404') {
+      // failed
```
* [x] Let return a message instead of just writing a comment, this will help you in debugging. You could return a customized status code ie,
`let response = {};`
`response={status: 404};`

Forison commented on src/modules/get-background.js:
```diff
@@ -0,0 +1,19 @@
+      changeBackground(background.hits[Math.floor(Math.random() * len)].largeImageURL);
+    }
+  } catch {
+    // errors
```
* [x] Let return a message instead of just writing a comment, this will help you in debugging. You could return a customized status code ie,
`let response = {};`
`response={status: 404};`

Forison commented on src/modules/get-weather-data.js:
```diff
@@ -0,0 +1,23 @@
+      getBackground(weatherData.weather[0].main);
+    }
+  } catch {
+    // error
```
* [x] Let return a message instead of just writing a comment, this will help you in debugging. You could return a customized status code ie,
`let response = {};`
`response={status: 404};`

Forison commented on src/modules/populate-data.js:
```diff
@@ -0,0 +1,130 @@
+  weatherDescription.textContent = `${data.weather[0].main}, ${data.weather[0].description}`;
+  weatherDescription.classList.add('weather-description');
+
+  // console.log(data);
```
* [x] Kindly remove all commented code

Forison commented on src/modules/populate-data.js:
```diff
@@ -0,0 +1,130 @@
+    Sunset: `${format(sunsetTime, 'p')}`,
+  };
+
+  // console.log(params.Visibility);
```
* [x] Kindly remove commented code

Forison commented on src/modules/populate-data.js:
```diff
@@ -0,0 +1,130 @@
+  weatherDataContainer.append(locationContainer, innerContainer, variousParamContainer);
+  container.appendChild(weatherDataContainer);
+
+  // console.log(conversion(data.main.temp));
```
* [x] Kindly remove all commented code

Forison reviewed (changes requested):
## Status::Changes Required❌
# 🙋‍♂️ @ershadul1 
#### You have done a good job on this one so far, your page looks very nice, the readme looks very professional and the code is well organized😇.
Here are a few changes we have to make to get the project approved.
* [x] Let us remove all commented code from our code.
* [x] Let us improve how we handle our promise
* [x] Necessary run common commands like `build`, `development` should be in `package.json` inside \`scripts'. Here is a basic example ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/scripts_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622133/))
- Kindly check the comments below for details about the suggestions above
#### That is it kindly make the changes and submit a re-review😎
#### happy coding👨‍💻
Addo Forison

ershadul1 committed:
Remove commented codes

ershadul1 committed:
Add scripts for dev and start

ershadul1 committed:
Add error handling in try catch

ershadul1 committed:
Built using webpack

ershadul1 commented:
### Changes made

- [X] Removed all commented codes.
- [X] Added scripts for dev and start in package.json.
- [X] Added error handling messages in async functions.

uche-inyama reviewed (approved):
## Approved

### Hi @ershadul1 :wave: 

Go ahead with your next task.

#### ``` Have a great day ``` :slightly_smiling_face: 

``` Uchechukwu Henry Inyama ```
